ouroboros-consensus/src/Ouroboros/Consensus/Node/NetworkProtocolVersion.hs

```diff
@@ -44,7 +45,7 @@ class HasNetworkProtocolVersion blk => SupportedNetworkProtocolVersion blk where
 
   -- | Enumerate all supported node-to-client versions
   supportedNodeToClientVersions
-    :: Proxy blk -> Map NodeToClientVersion (BlockNodeToClientVersion blk)
+    :: Proxy blk -> Map NodeToClientVersion (QueryVersion, BlockNodeToClientVersion blk)
```

Comment:
Just as with my other comment on `SerialiseNodeToClient`, I'm also thinking that this class shouldn't be related to `QueryVersion`.

This one is a bit murkier, I think, because of the kind of inconsistent way we're reusing the single `NodeToClientVersion` value for multiple `blk` types. We're very much evolving `NodeToClientVersion` itself as we add features on `mainnet`, ie `blk ~ CardanoBlock`. But we have instances for older block types too, eg `ByronBlock`. This is a mishmash, which is likely only fine so far because we're only ever really running this code with non-trivial versioning scenarios when `blk` is in fact `CardanoBlock`.

Anyhow: since, as I argued for `SerialiseNodeToClient`, no NTC serializations will depend on the `QueryVersion` except the serialization of `Query` itself, I think we can just have the `Query` serialization be the only place that `QueryVersion` is used, and moreover it can internally calculate the `QueryVersion` value as a direct `blk`-independent function of `NodeToClientVersion`.

Does this sound worth it? I can try to further sell both of these changes to you by pointing out that so many of the source files in this diff would no longer need any changes if we don't change either of these classes :P